include/Dialect/Btor/IR/BtorOps.td

```diff
@@ -524,10 +525,10 @@ def CmpOp : Btor_Op<"cmp", [NoSideEffect, SameTypeOperands,
 
   let arguments = (ins
       BtorPredicateAttr:$predicate,
-      SignlessIntegerLike:$lhs,
-      SignlessIntegerLike:$rhs
+      Btor_BitVec:$lhs,
+      Btor_BitVec:$rhs
   );
-  let results = (outs BoolLike:$result);
```

Comment:
Whenever you change a type of BoolLike, it's important to verify that the resulting Btor_BitVec has length 1. This can be done in the verifier.

Reply:
Added in verifier.
The builder itself is also custom specified such that the result is always a bit vector of length 1,